### PR TITLE
 Implement C and Swift API additions for shouldResolveCycle

### PR DESCRIPTION
**Pull request by dmbryson**

### CODE REVIEW

**Comment by dmbryson** [comment]
This PR should become a lot simpler once the prior cycle resolution change has been merged.

File: products/libllbuild/BuildSystem-C-API.cpp [hunk 1]
```diff
@@ -336,58 +336,112 @@ class CAPIBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
...
+      for (std::vector<core::Rule*>::const_iterator it = items.begin(); it != items.end(); ++it) {
+        core::Rule* rule = *it;
+        auto key = BuildKey::fromData(rule->key);
+        rules[idx++] = convertBuildKey(key);
+      }
+    }
+
+    ~CAPIRulesVector() {
+      for (auto& rule : rules) {
+        free((char *)rule.key);
```
Comment (ddunbar) [inline products/libllbuild/BuildSystem-C-API.cpp]: 👍

**Comment by ddunbar** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Pull request merged**